pci.cpp

```diff
@@ -93,6 +95,12 @@ int32 PciHandle::read32(uint64 offset, uint32 * value)
         req.bytes = sizeof(uint32);
         req.reg = (ULONG)offset;
 
+        if (!AllowPCICFGAccess(device, (DWORD)offset))
```

Comment:
same here

Reply:
also this check should be independent of the driver (put it in the beginning of the function)